core/src/main/scala/org/apache/spark/rdd/ReliableCheckpointRDD.scala

```diff
@@ -169,14 +174,24 @@ private[spark] object ReliableCheckpointRDD extends Logging {
     val bufferSize = env.conf.getInt("spark.buffer.size", 65536)
 
     val fileOutputStream = if (blockSize < 0) {
-      fs.create(tempOutputPath, false, bufferSize)
+      val checkpointCodec = env.conf.get("spark.checkpoint.compress.codec", "none")
+      val fileStream = fs.create(tempOutputPath, false, bufferSize)
+      if (CompressionCodec.ALL_COMPRESSION_CODECS_SHORT.contains(checkpointCodec)) {
+        val compressionCodec = CompressionCodec.createCodec(env.conf, checkpointCodec)
+        logInfo(s"Compressing using $checkpointCodec.")
+        compressionCodec.compressedOutputStream(fileStream)
+      } else {
+        fileStream
+      }
     } else {
       // This is mainly for testing purpose
       fs.create(tempOutputPath, false, bufferSize,
         fs.getDefaultReplication(fs.getWorkingDirectory), blockSize)
     }
     val serializer = env.serializer.newInstance()
     val serializeStream = serializer.serializeStream(fileOutputStream)
+    logInfo(s"Starting to write to checkpoint file $tempOutputPath.")
```

Comment:
This will make the logs verbose.
If it does help with debugging, you could make it logTrace - or remove it entirely.